src/page/MainScreen/MainScreen.test.tsx

```diff
@@ -1,21 +1,29 @@
 import React from 'react';
-import {render} from '@testing-library/react-native';
+import {fireEvent, render, waitFor} from '@testing-library/react-native';
 import {mockListMovies} from '../../mock/MockData';
 import {useNavigation} from '@react-navigation/native';
 import {useQuery, UseQueryResult} from '@tanstack/react-query';
 import {getMovie} from '../../utils/api/apiMovies';
 import MainScreen from './MainScreen';
+import {MovieDeckProps} from '../../components/MovieDeck/MovieDeck';
+import * as MovieDeck from '../../components/MovieDeck/MovieDeck.tsx';
 
 jest.mock('@tanstack/react-query');
 const mockedUseQuery = jest.mocked(useQuery);
 
+let mockedMovieDeck = jest.spyOn(MovieDeck, 'default');
+
 jest.mock('../../utils/api/apiMovies');
 const mockApi = jest.mocked(getMovie);
 
 jest.mock('@react-navigation/native');
 jest.mocked(useNavigation).mockReturnValue({goBack: jest.fn()});
 
 describe('MainScreen', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
```

Comment:
Look into this setting also, maybe it would easier for you https://jestjs.io/docs/configuration#clearmocks-boolean
In current setup it's a good solution 👍